### PR TITLE
fix: retain subtitles spanning HLS segment boundaries

### PR DESCRIPTION
**Pull request by LiHRaM**
## Summary

- Fix `FilterEvents` in `SubtitleEncoder` dropping subtitles that span an HLS segment boundary
- The `SkipWhile` predicate used `||` (OR), which dropped any subtitle that merely *started* before the segment — even if it was still on screen. Changed to `&&` (AND) so subtitles are only skipped when fully elapsed
- This bug has been present since the initial port from Emby in December 2018

Partially addresses #2547

### CODE REVIEW

File: MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs
Comment (gnattu) [inline MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs]: Although HLS webvtt may be not affected in this path, this will cause negative timestamp in other code paths.

**Review by gnattu — COMMENTED** [review]
IIRC most clients do not ask the server for subtitle transcoding (maybe android tv is the most popular one that does) so this might not be as helpful as it seems.

And it is also weird for me that you believe the old behavior would cause desync. The old behavior will drop cues at the segment boundary, which is closer to missing subtitles rather than timestamp shifting that the users originally reported. 

The old behavior is still a bug though, but this is a minor fix which cannot address the timestamp shifting issue.

**Comment by LiHRaM** [comment]
Point taken! I'll update the description - my thinking was that this contributes to the overall poor UX around subtitle desync, but it was misleading to say it _causes_ desync, as it's just a boundary condition.

**Review by gnattu — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

File: tests/Jellyfin.MediaEncoding.Tests/Subtitles/FilterEventsTests.cs [hunk 1]
```diff
@@ -0,0 +1,282 @@
+using System;
+using AutoFixture;
+using AutoFixture.AutoMoq;
+using MediaBrowser.MediaEncoding.Subtitles;
+using MediaBrowser.Model.MediaInfo;
+using Xunit;
+
+namespace Jellyfin.MediaEncoding.Subtitles.Tests
```
Comment (Bond-009) [inline tests/Jellyfin.MediaEncoding.Tests/Subtitles/FilterEventsTests.cs:8]: use file scoped namespace

**Review by Bond-009 — APPROVED** [review]
